assets/components/paymentButtons/stripePopUpButton/stripePopUpButton.jsx

```diff
@@ -51,6 +52,15 @@ const StripePopUpButton = (props: PropTypes) => (
   />
 );
 
+function formIsValid(formClassName: string) {
+  const form = document.getElementsByClassName(formClassName);
+  let isValid = true;
+  if (form && form.length > 0) {
+    const formElements = [...form[0].getElementsByTagName('input')];
+    isValid = formElements && formElements.reduce((acc, el) => acc && el.validity.valid, true);
```

Comment:
also as discussed with Jonny (and related to my longer comment): the simplicity and generality of this shows how nice having a uniform interface for checking constraints can be